### PR TITLE
deps: Upgrade to react 18, fix hydration error

### PR DESCRIPTION
**Pull request by BRKalow**
## 🔗 Relevant links

- [Preview link](https://dev-portal-git-BRANCH_NAME-hashicorp.vercel.app/PATH_TO_VIEW) 🔎
- [Asana task](url) 🎟️

## 🗒️ What

- Upgrade react to v18
- Update testing-library dependencies to support v18
- Adjust `VideoEmbed` to render a static placeholder during SSR to avoid hydration mismatch errors
- Update tests as a result of testing dependency breaking changes

### CODE REVIEW

File: src/contexts/__tests__/current-product.test.tsx [hunk 1]
```diff
@@ -1,6 +1,6 @@
-import { ReactNode } from 'react'
+import { ReactNode, Component } from 'react'
 import { render, screen } from '@testing-library/react'
-import { renderHook } from '@testing-library/react-hooks'
```
Comment (BRKalow) [inline src/contexts/__tests__/current-product.test.tsx:3]: This library got included in `@testing-library/react` as part of the changes to support react 18, so making the switch.

File: src/contexts/__tests__/current-product.test.tsx [hunk 2]
```diff
@@ -48,11 +48,32 @@ describe('CurrentProductContext', () => {
 	})
 
 	test('useCurrentProduct throws an error if not used within CurrentProductProvider', () => {
-		const { result } = renderHook(() => useCurrentProduct())
+		const spy = jest.spyOn(console, 'error').mockImplementation(() => void 0)
 
-		expect(result.error.message).toBe(
```
Comment (BRKalow) [inline src/contexts/__tests__/current-product.test.tsx:53]: `result.error` no longer exists, so I had to add some code to capture the error and silence the console.

**Review by zchsh — APPROVED** [review]
✅ This looks good to me!

I poked around the preview, on pages with videos, as a rough way to validate that the changes to `VideoEmbed` are all good. Happy to dive in further there if there's more I could do in terms of testing 💖

**Review by kendallstrautman — APPROVED** [review]
Messed around on the preview and everything looks good! 👏 Thanks for doing this.

**Pull request merged**